src/sql/src/plan/statement/ddl.rs

```diff
@@ -955,10 +955,32 @@ pub fn plan_create_source(
                         }
                     };
 
-                    let mir_cast = cast.lower_uncorrelated().expect(
-                        "lower_uncorrelated should not fail given that there is no correlation \
-                            in the input col_expr",
-                    );
+                    // We expect only reg* types to encounter this issue. Users
+                    // can ingest the data as text if they need to ingest it.
+                    // This is acceptable because we don't expect the OIDs from
+                    // an external PG source to be unilaterally usable in
+                    // resolving item names in MZ.
+                    let mir_cast = cast.lower_uncorrelated().map_err(|_e| {
```

Comment:
Handling the error gracefully is a nice way of protecting ourselves from this in the future! 🙌🏽